nautobot/ipam/views.py

```diff
@@ -672,7 +672,13 @@ class PrefixBulkDeleteView(generic.BulkDeleteView):
 
 
 class IPAddressListView(generic.ObjectListView):
-    queryset = IPAddress.objects.select_related("tenant", "status", "role")
+    queryset = IPAddress.objects.select_related("tenant", "status", "role").annotate(
+        interface_count=Count("interfaces"),
+        interface_parent_count=(Count("interfaces__device", distinct=True)),
+        vm_interface_count=Count("vm_interfaces"),
+        vm_interface_parent_count=(Count("vm_interfaces__virtual_machine", distinct=True)),
+        assigned_count=Count("interfaces") + Count("vm_interfaces"),
```

Comment:
SQL for this annotation

```sql
SELECT 
  "ipam_ipaddress"."id", 
  "ipam_ipaddress"."created", 
  "ipam_ipaddress"."last_updated", 
  "ipam_ipaddress"."_custom_field_data", 
  "ipam_ipaddress"."host", 
  "ipam_ipaddress"."mask_length", 
  "ipam_ipaddress"."type", 
  "ipam_ipaddress"."status_id", 
  "ipam_ipaddress"."role_id", 
  "ipam_ipaddress"."parent_id", 
  "ipam_ipaddress"."ip_version", 
  "ipam_ipaddress"."tenant_id", 
  "ipam_ipaddress"."nat_inside_id", 
  "ipam_ipaddress"."dns_name", 
  "ipam_ipaddress"."description", 
  COUNT(
    "ipam_ipaddresstointerface"."interface_id"
  ) AS "interface_count", 
  COUNT(
    DISTINCT "dcim_interface"."device_id"
  ) AS "interface_parent_count", 
  COUNT(
    "ipam_ipaddresstointerface"."vm_interface_id"
  ) AS "vm_interface_count", 
  COUNT(
    DISTINCT "virtualization_vminterface"."virtual_machine_id"
  ) AS "vm_interface_parent_count", 
  (
    COUNT(
      "ipam_ipaddresstointerface"."interface_id"
    ) + COUNT(
      "ipam_ipaddresstointerface"."vm_interface_id"
    )
  ) AS "assigned_count", 
  "extras_status"."id", 
  "extras_status"."created", 
  "extras_status"."last_updated", 
  "extras_status"."_custom_field_data", 
  "extras_status"."name", 
  "extras_status"."color", 
  "extras_status"."description", 
  "extras_role"."id", 
  "extras_role"."created", 
  "extras_role"."last_updated", 
  "extras_role"."_custom_field_data", 
  "extras_role"."name", 
  "extras_role"."color", 
  "extras_role"."description", 
  "extras_role"."weight", 
  "tenancy_tenant"."id", 
  "tenancy_tenant"."created", 
  "tenancy_tenant"."last_updated", 
  "tenancy_tenant"."_custom_field_data", 
  "tenancy_tenant"."name", 
  "tenancy_tenant"."tenant_group_id", 
  "tenancy_tenant"."description", 
  "tenancy_tenant"."comments" 
FROM 
  "ipam_ipaddress" 
  LEFT OUTER JOIN "ipam_ipaddresstointerface" ON (
    "ipam_ipaddress"."id" = "ipam_ipaddresstointerface"."ip_address_id"
  ) 
  LEFT OUTER JOIN "dcim_interface" ON (
    "ipam_ipaddresstointerface"."interface_id" = "dcim_interface"."id"
  ) 
  LEFT OUTER JOIN "virtualization_vminterface" ON (
    "ipam_ipaddresstointerface"."vm_interface_id" = "virtualization_vminterface"."id"
  ) 
  INNER JOIN "extras_status" ON (
    "ipam_ipaddress"."status_id" = "extras_status"."id"
  ) 
  LEFT OUTER JOIN "extras_role" ON (
    "ipam_ipaddress"."role_id" = "extras_role"."id"
  ) 
  LEFT OUTER JOIN "tenancy_tenant" ON (
    "ipam_ipaddress"."tenant_id" = "tenancy_tenant"."id"
  ) 
GROUP BY 
  "ipam_ipaddress"."id", 
  "extras_status"."id", 
  "extras_role"."id", 
  "tenancy_tenant"."id"

```

Reply:
As long as pagination is kept to a reasonable amount, this should not get too out of hand, especially since it's just doing all the joins for doing counts.